### PR TITLE
[#3266] Update text of relation types

### PR DESCRIPTION
**Pull request by Maurier**
Under References, the following relation types have been updated to human-friendly strings:

- "isVersionOf" now reads "This resource updates and replaces previous version"
- "isReplacedBy" now reads "This resource has been replaced by newer version"

### Pull Request Checklist: 
- [x] Positive Test Case Written by Dev

- [ ] Automated Testing

- [ ] Sufficient User and Developer Documentation

- [x] Passing Jenkins Build

- [ ] Peer Code review and approval

### Positive Test Case
1. Create a resource.
2. Create a new version of it.
3. Check both resources' relations that were automatically created.
4. Verify that the strings used as relation types are human-friendly.

### CODE REVIEW

File: theme/templates/resource-landing-page/references.html [hunk 1]
```diff
@@ -45,6 +45,10 @@ <h4>Related Resources</h4>
                                     This resource is described by:
                                 {% elif relation.type == "IsReferencedBy" %}
                                     This resource is referenced by:
+                                {% elif relation.type == "isVersionOf" %}
```
Comment (pkdash) [inline theme/templates/resource-landing-page/references.html]: If we implement a new property (display_type) for the Relation class, we don't have to do all these if...elifs in the template. We can have one line {{ relation.display_type }}. Just a thought.

**Comment by cuahsimarko** [comment]
While we're in here, I'd like to make some suggestions. The test could be easier to digest and we should ensure Cancel is the default action:
image

New Resource Version
Creating a new version will obsolete this resource.

- Obsoleted resource will be still available in HydroShare
- The new version is a complete new resource with new identifier
- Only the newer version will be Discoverable in Search
- Old and new versions will link to each other
- This can be repeated resulting in a sequence of multiple versions each linked to next and previous versions
- Only the most recent version can be edited or deleted
- If you delete the most recent version, the previous version becomes the most recent, therefore the only editable one

**Comment by cuahsimarko** [comment]
@dtarb do you like this?

**Comment by dtarb** [comment]
This text is pretty wordy, as was the old text.  Someone with a flair for concise communication needs to put thought into this.  I worry that if we have to explain so much on the interface, that we have a complicated concept. Until I, or someone else can come up with better wording, I am OK with it.

**Comment by horsburgh** [comment]
Could we just keep the first two sentences and have a "Learn more" link to the help files that explain more about versioning?

**Comment by cuahsimarko** [comment]
guided workflows discussion https://github.com/hydroshare/hydroshare/issues/1653

**Comment by cuahsimarko** [comment]
Per agreement with Mauriel moving scope creep to https://github.com/hydroshare/hydroshare/issues/3334 and merging PR

**Pull request merged**